### PR TITLE
Form 21-8940: Ability to populate Doctor’s care interview questions#14531

### PR DESCRIPTION
**Pull request by zurbergram**
## Description
As a Veteran submitting a claim for Individual Unemployability on the Vets.gov platform, I need the ability to populate interview questions regarding Doctor’s care, so that I can provide additional evidence associated to my Individual Unemployability while completing the Form 21-8940 application interview.

## Testing done
- [x] Unit testing
- [x] local testing

## Screenshots
screen shot 2019-01-07 at 4 03 39 pm
screen shot 2019-01-07 at 4 03 45 pm

## Acceptance criteria
- [x] The page content displays as:

        Please tell us where and when you were treated by a doctor.

        Doctor’s name { text field }

        Doctor’s address { address pattern }

        When were you under this doctor's care? Include the dates for the whole time you were under 
        this doctor’s care. { text box }

        Add Another { button }

        If you received care from a private medical facility, we can get that information for you. We'll ask 
        you about that later in this process.

        If you received care at a VA medical facility, we have access to those records and can get them 
        for you.

        [ Back | Continue ] { buttons } Finish this application later.

        { end if doctor’s care }
- [x] The user is able to move to the next step in the process, Story#14532: Form 21-8940: Ability to populate Disability Dates interview questions
- [x] Field validations (if applicable per the linked document for Validations above) are enforced in the UI
- [x] Error messages (if applicable per the linked document for Validations above) are displayed in the UI

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by imeecuison — APPROVED** [review]
LGTM!

**Comment by zurbergram** [comment]
@department-of-veterans-affairs/code-reviewers-for-external-teams-group Could I get a code review on this at your earliest convenience. Thank you!

File: src/applications/disability-benefits/all-claims/pages/index.js [hunk 1]
```diff
@@ -1,755 +1,171 @@
...
-} from './physicalHealthChanges';
-
-import {
-  uiSchema as hospitalizationHistoryUISchema,
-  schema as hospitalizationHistorySchema,
-} from './hospitalizationHistory';
-
-import * as medicalCare from './medicalCare';
-
+import * as alternateNames from './alternateNames';
```
Comment (jbalboni) [inline src/applications/disability-benefits/all-claims/pages/index.js]: The changes in this file look like refactoring that's unrelated to the changes you're making, which means they should probably be in a separate PR and discussed with @markgreenburg and @cvalarida.

File: src/applications/disability-benefits/all-claims/config/8940/index.js [hunk 2]
```diff
@@ -63,6 +65,13 @@ export default function() {
         schema: hospitalizationHistory.schema,
       },
       // 8940 - Doctor Treatment
+      unemployabilityDoctorCare: {
+        title: "Doctor's care",
```
Comment (jbalboni) [inline src/applications/disability-benefits/all-claims/config/8940/index.js]: This should use an apostrophe (`’`) instead of a `'` mark. There are several instances of that in this PR, can you go through and update them?

File: src/applications/disability-benefits/all-claims/pages/unemployabilityDoctorCare.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+          },
+          addressLine2: {
+            'ui:title': 'Street address (line 2)',
+          },
+          state: {
+            'ui:title': 'State',
+          },
+          zipCode: {
+            'ui:title': 'Postal Code',
+            'ui:validations': [validateZIP],
```
Comment (jbalboni) [inline src/applications/disability-benefits/all-claims/pages/unemployabilityDoctorCare.js]: Is this going to work if someone chooses a non-USA country and tries to enter a postal code in a different format?

Comment (zurbergram) [reply src/applications/disability-benefits/all-claims/pages/unemployabilityDoctorCare.js]: If zip is not valid returns error message "Please enter a valid 5 or 9 digit Postal code (dashes allowed)", so if international zip codes are 9 or less they should work

Comment (jbalboni) [reply src/applications/disability-benefits/all-claims/pages/unemployabilityDoctorCare.js]: I think that validation function checks for 5 or 9 digit codes, so in between those counts wouldn't work. Also, Canadian postal codes contain letters, not just digits.

The zip code logic for different countries is handled by default in that addressUI definition, if the `postalCode` property is used. Is it necessary to switch it to `zipCode`?

Comment (jbalboni) [reply src/applications/disability-benefits/all-claims/pages/unemployabilityDoctorCare.js]: Deferred this issue to https://github.com/department-of-veterans-affairs/vets-website/issues/9367

**Review by jbalboni — APPROVED** [review]
This looks good, aside from the issue mentioned around zip codes, which appears to be a problem in other places as well.
